File: puzzles/go/the-descent.go
package main

import "fmt"

func main() {
  for {
    var max = -1
    var indexMax int
    for i := 0; i < 8; i++ {
      var height int // represents the height of one mountain.
      fmt.Scan(&height)
      if height > max {
        max = height
        indexMax = i
      }
    }
    
    fmt.Println(indexMax) // The index of the mountain to fire on.
  }
}

File: puzzles/kotlin/src/the-descent.kt
import java.util.Scanner

fun main(args : Array<String>) {
  val input = Scanner(System.`in`)

  // game loop
  while (true) {
    var max = -1
    var indexMax = 0
    for (i in 0 until 8) {
      val height = input.nextInt() // represents the height of one mountain.
      if (height > max) {
        max = height
        indexMax = i
      }
    }

    println(indexMax) // The index of the mountain to fire on.
  }
}

